var/spack/repos/builtin/packages/r-oligoclasses/package.py

```diff
@@ -7,25 +7,36 @@
 
 
 class ROligoclasses(RPackage):
-    """This package contains class definitions, validity checks,
-    and initialization methods for classes used by the oligo and
-    crlmm packages."""
+    """Classes for high-throughput arrays supported by oligo and crlmm.
 
-    homepage = "https://www.bioconductor.org/packages/oligoClasses/"
+       This package contains class definitions, validity checks, and
+       initialization methods for classes used by the oligo and crlmm
+       packages."""
+
+    homepage = "https://bioconductor.org/packages/oligoClasses"
     git      = "https://git.bioconductor.org/packages/oligoClasses.git"
 
+    version('1.46.0', commit='325684f66fc92f778098f24bcfbef0ce3da9717c')
+    version('1.44.0', commit='d3e1134cdbea5f95b83215dc66e5f7b6a1cd0638')
+    version('1.42.0', commit='ef125700d487b470281a9c1e985390633c4dd2bd')
+    version('1.40.0', commit='32f40617e62d05c457baaebc7e27585b852848ed')
     version('1.38.0', commit='fe2bb7f02c7ed3cbd338254c27ceba6ff829a962')
 
-    depends_on('r@3.4.0:3.4.9', when='@1.38.0')
-    depends_on('r-biocgenerics', type=('build', 'run'))
-    depends_on('r-biobase', type=('build', 'run'))
-    depends_on('r-iranges', type=('build', 'run'))
-    depends_on('r-genomicranges', type=('build', 'run'))
+    depends_on('r@2.14:', type=('build', 'run'))
+    depends_on('r-biocgenerics@0.3.2:', type=('build', 'run'))
+    depends_on('r-biobase@2.17.8:', type=('build', 'run'))
+    depends_on('r-iranges@2.5.17:', type=('build', 'run'))
+    depends_on('r-genomicranges@1.23.7:', type=('build', 'run'))
     depends_on('r-summarizedexperiment', type=('build', 'run'))
-    depends_on('r-biostrings', type=('build', 'run'))
-    depends_on('r-affyio', type=('build', 'run'))
+    depends_on('r-biostrings@2.23.6:', type=('build', 'run'))
+    depends_on('r-affyio@1.23.2:', type=('build', 'run'))
     depends_on('r-ff', type=('build', 'run'))
     depends_on('r-foreach', type=('build', 'run'))
     depends_on('r-biocinstaller', type=('build', 'run'))
```

Comment:
I think this dep was replaced with `r-biocmanager`, so it is no longer needed in newer versions.

Reply:
Ping

Reply:
fixed